sympy/physics/continuum_mechanics/beam.py

```diff
@@ -0,0 +1,567 @@
+
+        bc_eqs = []
+        for position, value in self._boundary_conditions['slope']:
+            eqs = slope_curve.subs(x, position) - value
+            bc_eqs.append(eqs)
+
+        constants = list(linsolve(bc_eqs, C3))
+        slope_curve = slope_curve.subs({C3 : constants[0][0]})
+        return S(1)/(E*I)*slope_curve
+
+
+    def deflection(self):
+        """
+        Returns a Singularity Function expression which represents
+        the elastic curve or deflection of the Beam object.
+
+        Examples
+        ========
+        >>> from sympy.physics.continuum_mechanics.beam import Beam, DistributedLoad, PointLoad
+        >>> from sympy import Symbol
+        >>> from sympy.physics.mechanics import Point
+        >>> E = Symbol('E')
+        >>> I = Symbol('I')
+        >>> b = Beam(4, E, I)
+        >>> P1 = Point('0')
+        >>> P2 = Point('2')
+        >>> P3 = Point('3')
+        >>> Load_1 = PointLoad(location = P1, value = -3, moment = True)
+        >>> Load_2 = PointLoad(location = P2, value = 4)
+        >>> Load_3 = DistributedLoad(start = P3, order = 2, value = -2)
+        >>> b.apply_loads(Load_1, Load_2, Load_3)
+        >>> b.apply_boundary_conditions(moment = [(0, 4), (4, 0)], deflection = [(0, 2)], slope = [(0, 1)])
+        >>> b.deflection()
+        (-71*x**3/144 + 7*x**2/2 + x - 3*SingularityFunction(x, 0, 2)/2 + 2*SingularityFunction(x, 2, 3)/3 - SingularityFunction(x, 3, 6)/180 + 2)/(E*I)
+        """
+        x = Symbol('x')
+        E = self.elastic_modulus
+        I = self.second_moment
+        if not self._boundary_conditions['deflection'] and not self._boundary_conditions['slope']:
+            return S(1)/(E*I)*integrate(integrate(self.bending_moment(), x), x)
+        elif not self._boundary_conditions['deflection']:
+            return integrate(self.slope(), x)
+        elif not self._boundary_conditions['slope'] and self._boundary_conditions['deflection']:
+            C3 = Symbol('C3')
+            slope_curve = integrate(self.bending_moment(), x) + C3
+            deflection_curve = integrate(slope_curve, x)
+            bc_eqs = []
+            for position, value in self._boundary_conditions['deflection']:
+                eqs = deflection_curve.subs(x, position) - value
+                bc_eqs.append(eqs)
+            constants = list(linsolve(bc_eqs, C3))
+            deflection_curve = deflection_curve.subs({C3 : constants[0][0]})
+            return S(1)/(E*I)*deflection_curve
+
+        C4 = Symbol('C4')
+        deflection_curve = integrate((E*I)*self.slope(), x) + C4
+
+        bc_eqs = []
+        for position, value in self._boundary_conditions['deflection']:
+            eqs = deflection_curve.subs(x, position) - value
+            bc_eqs.append(eqs)
+
+        constants = list(linsolve(bc_eqs, C4))
+        deflection_curve = deflection_curve.subs({C4 : constants[0][0]})
+        return S(1)/(E*I)*deflection_curve
+
+
+class PointLoad(object):
```

Comment:
So, I'm not sure if it is worth designing for something that we don't even know if or how it will work.